.github/workflows/check-readme.yml

```diff
@@ -0,0 +1,42 @@
+name: Check README
+on:
+  push:
+    branches: [main]
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  check-readme:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            .github
+            packages
+            README.md
+            package.json
+            yarn.lock
```

Comment:
I believe direct root files are always included(?) https://github.com/actions/checkout#fetch-only-the-root-files-and-github-and-src-folder